fix(go) update doc to use go mod when installing go-pluginserver (#2787)
Kong/docs.konghq.com: app/enterprise/2.1.x/go.md

go build -buildmode plugin go-hello.go
```

> Note: when installing go-pluginserver binary globally user will need to

Reviewer: ```suggestion
Note: When installing the `go-pluginserver` binary globally, you  need to
```

> enable go.mod support by using:

Reviewer: ```suggestion
enable `go.mod` support:
```
